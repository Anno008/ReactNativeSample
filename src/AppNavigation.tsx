import React from "react";
// Eventually try out react-native-navigation, better performance and richer configuration
// requires setup
import {
  createAppContainer,
  createBottomTabNavigator,
  NavigationContainer,
  TabScene,
} from "react-navigation";

import { HomeIcon } from "components/icons/HomeIcon";
import { InfoIcon } from "components/icons/InfoIcon";
import { getMainNavigationIcon } from "decorators/GetMainNavigationIcon";
import AboutScreen from "screens/AboutScreen";
import HomeScreen from "screens/HomeScreen";

const AppNavigation: NavigationContainer = createBottomTabNavigator(
  {
    About: {
      navigationOptions: {
        tabBarIcon: (props: TabScene) => getMainNavigationIcon(InfoIcon, props),
        tabBarLabel: "About"
      },
      screen: AboutScreen
    },
    Home: {
      navigationOptions: {
        tabBarIcon: (props: TabScene) => getMainNavigationIcon(HomeIcon, props),
        tabBarLabel: "Home"
      },
      screen: HomeScreen
    }
  },
  // Router config
  {
    initialRouteName: "Home",
    order: ["Home", "About"],
    tabBarOptions: {
      // activeBackgroundColor: "rgba(107, 185, 240, 0.6)",
      activeTintColor: "navy",
      inactiveTintColor: "gray",
      labelStyle: {
        fontSize: 14
      },
      style: {
        backgroundColor: "rgba(107, 185, 240, 0.5)",
        height: 50
      }
    }
  }
);

export default createAppContainer(AppNavigation);
