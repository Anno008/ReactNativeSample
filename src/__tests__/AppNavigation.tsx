import AppNavigation from "AppNavigation";
import { shallow } from "enzyme";
import React from "react";

describe("AppNavigation tests", () => {
  it("should render", () => {
    const wrapper = shallow(<AppNavigation />);
    expect(wrapper).toMatchSnapshot();
  });
});
