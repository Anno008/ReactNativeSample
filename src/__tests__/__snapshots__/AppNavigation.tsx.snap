// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigation tests should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Navigator
        navigation={
          Object {
            "_childrenNavigation": Object {},
            "actions": Object {},
            "addListener": [Function],
            "dangerouslyGetParent": [Function],
            "dispatch": [Function],
            "getChildNavigation": [Function],
            "getScreenProps": [Function],
            "goBack": [Function],
            "isFocused": [Function],
            "navigate": [Function],
            "router": Object {
              "childRouters": Object {
                "About": null,
                "Home": null,
              },
              "getActionCreators": [Function],
              "getActionForPathAndParams": [Function],
              "getComponentForRouteName": [Function],
              "getComponentForState": [Function],
              "getPathAndParamsForState": [Function],
              "getScreenOptions": [Function],
              "getStateForAction": [Function],
            },
            "setParams": [Function],
            "state": Object {
              "index": 0,
              "isTransitioning": false,
              "routes": Array [
                Object {
                  "key": "Home",
                  "params": undefined,
                  "routeName": "Home",
                },
                Object {
                  "key": "About",
                  "params": undefined,
                  "routeName": "About",
                },
              ],
            },
          }
        }
      />,
      "value": Object {
        "_childrenNavigation": Object {},
        "actions": Object {},
        "addListener": [Function],
        "dangerouslyGetParent": [Function],
        "dispatch": [Function],
        "getChildNavigation": [Function],
        "getScreenProps": [Function],
        "goBack": [Function],
        "isFocused": [Function],
        "navigate": [Function],
        "router": Object {
          "childRouters": Object {
            "About": null,
            "Home": null,
          },
          "getActionCreators": [Function],
          "getActionForPathAndParams": [Function],
          "getComponentForRouteName": [Function],
          "getComponentForState": [Function],
          "getPathAndParamsForState": [Function],
          "getScreenOptions": [Function],
          "getStateForAction": [Function],
        },
        "setParams": [Function],
        "state": Object {
          "index": 0,
          "isTransitioning": false,
          "routes": Array [
            Object {
              "key": "Home",
              "params": undefined,
              "routeName": "Home",
            },
            Object {
              "key": "About",
              "params": undefined,
              "routeName": "About",
            },
          ],
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "navigation": Object {
          "_childrenNavigation": Object {},
          "actions": Object {},
          "addListener": [Function],
          "dangerouslyGetParent": [Function],
          "dispatch": [Function],
          "getChildNavigation": [Function],
          "getScreenProps": [Function],
          "goBack": [Function],
          "isFocused": [Function],
          "navigate": [Function],
          "router": Object {
            "childRouters": Object {
              "About": null,
              "Home": null,
            },
            "getActionCreators": [Function],
            "getActionForPathAndParams": [Function],
            "getComponentForRouteName": [Function],
            "getComponentForState": [Function],
            "getPathAndParamsForState": [Function],
            "getScreenOptions": [Function],
            "getStateForAction": [Function],
          },
          "setParams": [Function],
          "state": Object {
            "index": 0,
            "isTransitioning": false,
            "routes": Array [
              Object {
                "key": "Home",
                "params": undefined,
                "routeName": "Home",
              },
              Object {
                "key": "About",
                "params": undefined,
                "routeName": "About",
              },
            ],
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": undefined,
        "_currentValue2": undefined,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Navigator
          navigation={
            Object {
              "_childrenNavigation": Object {},
              "actions": Object {},
              "addListener": [Function],
              "dangerouslyGetParent": [Function],
              "dispatch": [Function],
              "getChildNavigation": [Function],
              "getScreenProps": [Function],
              "goBack": [Function],
              "isFocused": [Function],
              "navigate": [Function],
              "router": Object {
                "childRouters": Object {
                  "About": null,
                  "Home": null,
                },
                "getActionCreators": [Function],
                "getActionForPathAndParams": [Function],
                "getComponentForRouteName": [Function],
                "getComponentForState": [Function],
                "getPathAndParamsForState": [Function],
                "getScreenOptions": [Function],
                "getStateForAction": [Function],
              },
              "setParams": [Function],
              "state": Object {
                "index": 0,
                "isTransitioning": false,
                "routes": Array [
                  Object {
                    "key": "Home",
                    "params": undefined,
                    "routeName": "Home",
                  },
                  Object {
                    "key": "About",
                    "params": undefined,
                    "routeName": "About",
                  },
                ],
              },
            }
          }
        />,
        "value": Object {
          "_childrenNavigation": Object {},
          "actions": Object {},
          "addListener": [Function],
          "dangerouslyGetParent": [Function],
          "dispatch": [Function],
          "getChildNavigation": [Function],
          "getScreenProps": [Function],
          "goBack": [Function],
          "isFocused": [Function],
          "navigate": [Function],
          "router": Object {
            "childRouters": Object {
              "About": null,
              "Home": null,
            },
            "getActionCreators": [Function],
            "getActionForPathAndParams": [Function],
            "getComponentForRouteName": [Function],
            "getComponentForState": [Function],
            "getPathAndParamsForState": [Function],
            "getScreenOptions": [Function],
            "getStateForAction": [Function],
          },
          "setParams": [Function],
          "state": Object {
            "index": 0,
            "isTransitioning": false,
            "routes": Array [
              Object {
                "key": "Home",
                "params": undefined,
                "routeName": "Home",
              },
              Object {
                "key": "About",
                "params": undefined,
                "routeName": "About",
              },
            ],
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "navigation": Object {
            "_childrenNavigation": Object {},
            "actions": Object {},
            "addListener": [Function],
            "dangerouslyGetParent": [Function],
            "dispatch": [Function],
            "getChildNavigation": [Function],
            "getScreenProps": [Function],
            "goBack": [Function],
            "isFocused": [Function],
            "navigate": [Function],
            "router": Object {
              "childRouters": Object {
                "About": null,
                "Home": null,
              },
              "getActionCreators": [Function],
              "getActionForPathAndParams": [Function],
              "getComponentForRouteName": [Function],
              "getComponentForState": [Function],
              "getPathAndParamsForState": [Function],
              "getScreenOptions": [Function],
              "getStateForAction": [Function],
            },
            "setParams": [Function],
            "state": Object {
              "index": 0,
              "isTransitioning": false,
              "routes": Array [
                Object {
                  "key": "Home",
                  "params": undefined,
                  "routeName": "Home",
                },
                Object {
                  "key": "About",
                  "params": undefined,
                  "routeName": "About",
                },
              ],
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": undefined,
          "_currentValue2": undefined,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
